src/modules/job-manager/alloc.c

```diff
@@ -232,6 +234,32 @@ int cancel_request (struct alloc *alloc, struct job *job)
     return 0;
 }
 
+static void update_note (flux_t *h, struct job *job, flux_jobid_t id,
+                         json_t *note, char **note_str)
+{
+    if (note) {
+        if (job->note) {
+            if (json_object_update (job->note, note) < 0)
+                flux_log (h,
+                          LOG_ERR,
+                          "sched.alloc-response: id=%ju failed update note",
+                          (uintmax_t)id);
+        }
+        else {
+            if (!(job->note = json_incref (note)))
```

Comment:
was just thinking about this.  Is this safe or should this be a `json_copy()`?  B/c we might be updating this, we may want our own copy.  But the message's copy should disappear shortly.

Reply:
If this is coming from the message payload, IMHO it should be safe to just take a reference (as `json_object_update()` is doing) because the message is destroyed as soon as the handler returns, in the same thread.